core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -800,6 +800,11 @@ class ReplicaManager(val config: KafkaConfig,
       }
 
       logManager.truncateTo(partitionsToMakeFollower.map(partition => (new TopicAndPartition(partition), partition.getOrCreateReplica().highWatermark.messageOffset)).toMap)
+      partitionsToMakeFollower.foreach(partition => {
```

Comment:
Can you add a detailed description of the fix somewhere? It makes sense intuitively, but it would be good it you explained the impact of this.